andela/mmdp_site
# MMDP-630: Generate gap analysis report

**What does this PR do?**
As a stakeholder I should be able to view a gap analysis report and quickly identify local government areas that are missing out on essential services under specific sub themes and pillars. This can help guide my decision to bring services to places that need it the most. 
This PR creates an endpoint which can be used to fetch the said gap analysis report

**description of Task to be completed?**

- query the database for a list of all pillars, sub themes and focus areas
- query the database for a list of local government areas and the focus areas associated with them
- use the 2 data sets to identify gaps in the said local government areas

**How should this be manually tested?**

- `git fetch`
- `git checkout ft-gap-analysis-report-MMDP-630`
- `yarn dev`
-  use postman to hit `http://0.0.0.0:3000/api/v1/gapAnalysis` and you should see the gap analysis report generated

**Any background context you want to provide?**

N/A

**What are the relevant Jira issues?**

[MMDP-630](https://viisaus.atlassian.net/browse/MMDP-630)

**Screenshots (if appropriate)**
![image](https://user-images.githubusercontent.com/40663828/59921510-8ad6f980-9436-11e9-90e5-e7dd55ffc0dc.png)

![image](https://user-images.githubusercontent.com/40663828/59921550-b823a780-9436-11e9-9282-f4b35c937a43.png)


**Questions:**

N/A
